### PR TITLE
[incubator-kie-issues-1131] test migration from V7 to code generation-11

### PR DESCRIPTION
**Pull request by Abhitocode**
In addition to previous changes associated,
Migrated tests within StandaloneBPMNProcessTest.java

The test can be identified by referring to StandaloneBPMNProcessTest.java:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java

Closes https://github.com/apache/incubator-kie-issues/issues/1131.

**Note**: will resolve conflicts once reviewed for any changes.

### CODE REVIEW

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 1]
```diff
@@ -255,69 +320,73 @@ public void testExclusiveSplitXPath() throws Exception {
...
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("Email2", new SystemOutWorkItemHandler());
-        kruntime.signalEvent("No", "NoValue", processInstance.getStringId());
-        assertProcessInstanceCompleted(processInstance.getStringId(), kruntime);
+        org.kie.kogito.process.Process<EventBasedSplitModel> processDefinition = EventBasedSplitProcess.newProcess(app);
+        EventBasedSplitModel model = processDefinition.createModel();
+        org.kie.kogito.process.ProcessInstance<EventBasedSplitModel> instance = processDefinition.createInstance(model);
+        instance.start();
+        assertThat(instance).extracting(ProcessInstance::status).isEqualTo(ProcessInstance.STATE_ACTIVE);
+        ProcessTestHelper.registerHandler(app, "Email1", new SystemOutWorkItemHandler());
+        ProcessTestHelper.registerHandler(app, "Email2", new SystemOutWorkItemHandler());
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:334]: The handlers are already registered at the beginning of the test, no need to register them again IMO.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 2]
```diff
@@ -255,69 +320,73 @@ public void testExclusiveSplitXPath() throws Exception {
...
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("Email2", new DoNothingWorkItemHandler());
-        kruntime.signalEvent("No", "NoValue", processInstance.getStringId());
-        assertThat(processInstance.getState()).isEqualTo(KogitoProcessInstance.STATE_ACTIVE);
+        org.kie.kogito.process.Process<EventBasedSplitModel> processDefinition = EventBasedSplitProcess.newProcess(app);
+        EventBasedSplitModel model = processDefinition.createModel();
+        org.kie.kogito.process.ProcessInstance<EventBasedSplitModel> instance = processDefinition.createInstance(model);
+        instance.start();
+        assertThat(instance).extracting(ProcessInstance::status).isEqualTo(ProcessInstance.STATE_ACTIVE);
+        ProcessTestHelper.registerHandler(app, "Email1", new DoNothingWorkItemHandler());
+        ProcessTestHelper.registerHandler(app, "Email2", new DoNothingWorkItemHandler());
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:358]: The handlers are already registered at the beginning of the test, no need to register them again.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 3]
```diff
@@ -255,69 +320,73 @@ public void testExclusiveSplitXPath() throws Exception {
...
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("Email2", new DoNothingWorkItemHandler());
-        // No
-        kruntime.signalEvent("No", "NoValue", processInstance.getStringId());
+        org.kie.kogito.process.Process<EventBasedSplitModel> processDefinition = EventBasedSplitProcess.newProcess(app);
+        EventBasedSplitModel model = processDefinition.createModel();
+        org.kie.kogito.process.ProcessInstance<EventBasedSplitModel> instance = processDefinition.createInstance(model);
+        instance.start();
+        assertThat(instance).extracting(ProcessInstance::status).isEqualTo(ProcessInstance.STATE_ACTIVE);
+        ProcessTestHelper.registerHandler(app, "Email1", new SystemOutWorkItemHandler());
+        ProcessTestHelper.registerHandler(app, "Email2", new DoNothingWorkItemHandler());
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:383]: The handlers are already registered at the beginning of the test, no need to register them again.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 4]
```diff
@@ -255,69 +320,73 @@ public void testExclusiveSplitXPath() throws Exception {
...
+        instance.send(Sig.of("Yes", "YesValue"));
+        assertThat(instance).extracting(ProcessInstance::status).isEqualTo(ProcessInstance.STATE_ACTIVE);
+        ProcessTestHelper.registerHandler(app, "Email1", new SystemOutWorkItemHandler());
+        ProcessTestHelper.registerHandler(app, "Email2", new DoNothingWorkItemHandler());
+
+        instance.send(Sig.of("No", "NoValue"));
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:389]: There is no assert here that actually evaluates if this signal has an impact. Maybe at least check if the state of the process is still active?

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 5]
```diff
@@ -255,69 +320,73 @@ public void testExclusiveSplitXPath() throws Exception {
...
+        ProcessTestHelper.registerHandler(app, "Email1", new DoNothingWorkItemHandler());
+        ProcessTestHelper.registerHandler(app, "Email2", new DoNothingWorkItemHandler());
+        instance.send(Sig.of("No", "NoValue"));
+        assertThat(instance).extracting(ProcessInstance::status).isEqualTo(ProcessInstance.STATE_ACTIVE);
     }
 
     @Test
-    public void testEventBasedSplitAfter() throws Exception {
-        // signaling the other alternative after one has been selected should
-        // have no effect
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:305]: You might want to leave the comment in here to capture some additional information about the test.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java [hunk 6]
```diff
@@ -720,11 +804,15 @@ public void afterProcessStarted(ProcessStartedEvent event) {
...
-        kruntime.startProcess("SignalEndEvent", params);
+    public void testSignalEnd() {
+        Application app = ProcessTestHelper.newApplication();
+
+        org.kie.kogito.process.Process<SignalEndEventModel> processDefinition = SignalEndEventProcess.newProcess(app);
+        SignalEndEventModel model = processDefinition.createModel();
+        model.setX("MyValue");
+
+        org.kie.kogito.process.ProcessInstance<SignalEndEventModel> instance = processDefinition.createInstance(model);
+        instance.start();
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/StandaloneBPMNProcessTest.java:815]: This test lacks an assert on whether the end signal has actually been sent.

**Review by elguardian — APPROVED** [review]
LGTM, needs one more a approval and be green

**Review by martinweiler — APPROVED** [review]
LGTM, thanks @Abhitocode

**Pull request merged**